diff --git a/crates/language_model/src/settings.rs b/crates/language_model/src/settings.rs
index 275fcf0417..d074ae6548 100644
--- a/crates/language_model/src/settings.rs
+++ b/crates/language_model/src/settings.rs
@@ -14,6 +14,7 @@ use crate::{
         cloud::{self, ZedDotDevSettings},
         copilot_chat::CopilotChatSettings,
         google::GoogleSettings,
+        mlc_llm::MlcLlmSettings,
         ollama::OllamaSettings,
         open_ai::OpenAiSettings,
     },
@@ -61,6 +62,7 @@ pub struct AllLanguageModelSettings {
     pub zed_dot_dev: ZedDotDevSettings,
     pub google: GoogleSettings,
     pub copilot_chat: CopilotChatSettings,
+    pub mlc_llm: MlcLlmSettings,
 }
 
 #[derive(Default, Clone, Debug, Serialize, Deserialize, PartialEq, JsonSchema)]
@@ -72,6 +74,7 @@ pub struct AllLanguageModelSettingsContent {
     pub zed_dot_dev: Option<ZedDotDevSettingsContent>,
     pub google: Option<GoogleSettingsContent>,
     pub copilot_chat: Option<CopilotChatSettingsContent>,
+    pub mlc_llm: Option<MlcLlmSettingsContent>,
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, JsonSchema)]
@@ -230,6 +233,12 @@ pub struct ZedDotDevSettingsContent {
 #[derive(Default, Clone, Debug, Serialize, Deserialize, PartialEq, JsonSchema)]
 pub struct CopilotChatSettingsContent {}
 
+#[derive(Default, Clone, Debug, Serialize, Deserialize, PartialEq, JsonSchema)]
+pub struct MlcLlmSettingsContent {
+    pub api_url: Option<String>,
+    pub available_models: Option<Vec<provider::mlc_llm::AvailableModel>>,
+}
+
 impl settings::Settings for AllLanguageModelSettings {
     const KEY: Option<&'static str> = Some("language_models");
 
@@ -314,6 +323,12 @@ impl settings::Settings for AllLanguageModelSettings {
                     .as_ref()
                     .and_then(|s| s.available_models.clone()),
             );
+
+            // MLC-LLM
+            merge(
+                &mut settings.mlc_llm.api_url,
+                value.mlc_llm.as_ref().and_then(|s| s.api_url.clone()),
+            );
         }
 
         Ok(settings)
