diff --git a/crates/language_models/src/settings.rs b/crates/language_models/src/settings.rs
index f6602427cb..77be103f82 100644
--- a/crates/language_models/src/settings.rs
+++ b/crates/language_models/src/settings.rs
@@ -14,6 +14,7 @@ use crate::provider::{
     cloud::{self, ZedDotDevSettings},
     copilot_chat::CopilotChatSettings,
     google::GoogleSettings,
+    mlc_llm::MlcLlmSettings,
     ollama::OllamaSettings,
     open_ai::OpenAiSettings,
 };
@@ -59,6 +60,7 @@ pub struct AllLanguageModelSettings {
     pub zed_dot_dev: ZedDotDevSettings,
     pub google: GoogleSettings,
     pub copilot_chat: CopilotChatSettings,
+    pub mlc_llm: MlcLlmSettings,
 }
 
 #[derive(Default, Clone, Debug, Serialize, Deserialize, PartialEq, JsonSchema)]
@@ -70,6 +72,7 @@ pub struct AllLanguageModelSettingsContent {
     pub zed_dot_dev: Option<ZedDotDevSettingsContent>,
     pub google: Option<GoogleSettingsContent>,
     pub copilot_chat: Option<CopilotChatSettingsContent>,
+    pub mlc_llm: Option<MlcLlmSettingsContent>,
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, JsonSchema)]
@@ -228,6 +231,12 @@ pub struct ZedDotDevSettingsContent {
 #[derive(Default, Clone, Debug, Serialize, Deserialize, PartialEq, JsonSchema)]
 pub struct CopilotChatSettingsContent {}
 
+#[derive(Default, Clone, Debug, Serialize, Deserialize, PartialEq, JsonSchema)]
+pub struct MlcLlmSettingsContent {
+    pub api_url: Option<String>,
+    pub available_models: Option<Vec<provider::mlc_llm::AvailableModel>>,
+}
+
 impl settings::Settings for AllLanguageModelSettings {
     const KEY: Option<&'static str> = Some("language_models");
 
@@ -312,6 +321,12 @@ impl settings::Settings for AllLanguageModelSettings {
                     .as_ref()
                     .and_then(|s| s.available_models.clone()),
             );
+
+            // MLC-LLM
+            merge(
+                &mut settings.mlc_llm.api_url,
+                value.mlc_llm.as_ref().and_then(|s| s.api_url.clone()),
+            );
         }
 
         Ok(settings)
