diff --git a/crates/language_models/Cargo.toml b/crates/language_models/Cargo.toml
index 00d948bd2d..37041ac826 100644
--- a/crates/language_models/Cargo.toml
+++ b/crates/language_models/Cargo.toml
@@ -26,6 +26,7 @@ gpui.workspace = true
 http_client.workspace = true
 language_model.workspace = true
 menu.workspace = true
+mlc_llm = { workspace = true, features = ["schemars"] }
 ollama = { workspace = true, features = ["schemars"] }
 open_ai = { workspace = true, features = ["schemars"] }
 project.workspace = true
