diff --git a/crates/language_model/Cargo.toml b/crates/language_model/Cargo.toml
index e88675bbae..e05da4393f 100644
--- a/crates/language_model/Cargo.toml
+++ b/crates/language_model/Cargo.toml
@@ -35,6 +35,7 @@ http_client.workspace = true
 inline_completion_button.workspace = true
 log.workspace = true
 menu.workspace = true
+mlc_llm = { workspace = true, features = ["schemars"] }
 ollama = { workspace = true, features = ["schemars"] }
 open_ai = { workspace = true, features = ["schemars"] }
 parking_lot.workspace = true
