diff --git a/crates/language_model/Cargo.toml b/crates/language_model/Cargo.toml
index 0fc54d509d..fbab2acf5c 100644
--- a/crates/language_model/Cargo.toml
+++ b/crates/language_model/Cargo.toml
@@ -26,6 +26,7 @@ gpui.workspace = true
 http_client.workspace = true
 image.workspace = true
 log.workspace = true
+mlc_llm = { workspace = true, features = ["schemars"] }
 ollama = { workspace = true, features = ["schemars"] }
 open_ai = { workspace = true, features = ["schemars"] }
 parking_lot.workspace = true
